service/history/workflow/task_generator.go

```diff
@@ -108,6 +109,7 @@ type (
 		namespaceRegistry namespace.Registry
 		mutableState      MutableState
 		config            *configs.Config
+		archivalMetadata  archiver.ArchivalMetadata
```

Comment:
I am curious that if this can move to configs.Config?

Reply:
After several hours of trying, I determined this was too difficult